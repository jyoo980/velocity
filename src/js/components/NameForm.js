import { h } from "preact";

function NameForm({ onSubmit, visible }) {
  if (visible) {
    return null;
  }
  return (
    <div id="name-form">
      <label className="name-label" htmlFor="name">
        Hi. What's your name?
      </label>
      <input
        type="text"
        id="name"
        autoComplete="off"
        autoFocus
        number="50"
        onKeyPress={(e) => onSubmit(e)}
      />
    </div>
  );
}

export default NameForm;
