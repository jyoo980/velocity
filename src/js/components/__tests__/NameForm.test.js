import NameForm from "../NameForm";
import { h } from "preact";
import { mount } from "enzyme";

describe("NameForm", () => {
  test("NameForm", () => {
    const wrapper = mount(<NameForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
