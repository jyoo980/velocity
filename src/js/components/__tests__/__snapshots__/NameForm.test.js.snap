// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameForm NameForm 1`] = `
<NameForm>
  <div
    id="name-form"
  >
    <label
      className="name-label"
      htmlFor="name"
    >
      Hi. What's your name?
    </label>
    <input
      autoComplete="off"
      autoFocus={true}
      id="name"
      number="50"
      type="text"
    />
  </div>
</NameForm>
`;
